prod_api/tests/test_auth.py

```diff
@@ -0,0 +1,336 @@
+
+    :param issued_tokens: list of tokens provided by `issued_tokens` fixture
+    :type issued_tokens: list
+    :param prodapi_app: prod api app
+    :type prodapi_app: eve.flaskapp.Eve
+    :param prodapi_client: client for prod api
+    :type prodapi_client: flask.testing.FlaskClient
+    """
+
+    access_token_all_scopes = issued_tokens[0]['access_token']
+    access_token_no_scopes = issued_tokens[1]['access_token']
+
+    with prodapi_app.test_request_context():
+        for resource in RESOURCES:
+            # we send a requests with 'empty scopes' token
+            resp = prodapi_client.get(
+                url_for('{}|resource'.format(resource)),
+                headers={
+                    'Authorization': 'Bearer {}'.format(access_token_no_scopes)
+                }
+            )
+            resp_data = json.loads(resp.data.decode('utf-8'))
+            # we get a 401 response
+            assert resp.status_code == 401
+            assert resp_data == {
+                '_status': 'ERR',
+                '_error': {
+                    'code': 401,
+                    'message': 'Please provide proper credentials'
+                }
+            }
+            # we send a requests with 'all scopes' token
+            resp = prodapi_client.get(
+                url_for('{}|resource'.format(resource)),
+                headers={
+                    'Authorization': 'Bearer {}'.format(access_token_all_scopes)
+                }
+            )
+            # we get a 200 response
+            assert resp.status_code == 200
+
+
+@pytest.mark.parametrize('issued_tokens', [(("DESKS_READ",),)], indirect=True)
+def test_scopes_partial(issued_tokens, prodapi_app):
+    """
+    Send a request with DESKS_READ scope token to `desks` and `items`:
+        - make requests to `desks` and `items` endpoints
+        - ensure that only request to `desk` is authorized
+
+    :param prodapi_app: prod api app
+    :type prodapi_app: eve.flaskapp.Eve
+    :param issued_tokens: list of tokens provided by `issued_tokens` fixture
+    :type issued_tokens: list
+    """
+
+    access_token = issued_tokens[0]['access_token']
+    prodapi_client = prodapi_app.test_client()
+
+    with prodapi_app.test_request_context():
+        # we send a request with 'DESKS_READ scope' token to desks endpoint
+        resp = prodapi_client.get(
+            url_for('desks|resource'),
+            headers={
+                'Authorization': 'Bearer {}'.format(access_token)
+            }
+        )
+        # we get a 200 response
+        assert resp.status_code == 200
+
+        # we send a request with 'DESKS_READ scope' token to items endpoint
+        resp = prodapi_client.get(
+            url_for('archive|resource'),
+            headers={
+                'Authorization': 'Bearer {}'.format(access_token)
+            }
+        )
+        resp_data = json.loads(resp.data.decode('utf-8'))
+        # we get a 401 response
```

Comment:
same remark as above (either change to `403` or change the sequence diagram of auth server)